tui_cmds.go

```diff
@@ -25,21 +27,36 @@ const (
 	MethodDependabot mergeMethod = "@dependabot merge"
 )
 
-func mergePullRequest(pr pullRequest, method mergeMethod) tea.Cmd {
+type commander struct {
+	limiter *rate.Limiter
+}
+
+func newCommander() commander {
+	limiter := rate.NewLimiter(rate.Every(time.Second*2), 1)
+	return commander{
+		limiter: limiter,
+	}
+}
+
+func (c commander) mergePullRequest(pr pullRequest, method mergeMethod) tea.Cmd {
 	return func() tea.Msg {
 		switch method {
 		case MethodRebase:
 			fallthrough
 		case MethodMerge:
 			fallthrough
 		case MethodSquash:
+			r := c.limiter.Reserve()
```

Comment:
why not just `c.limiter.Wait(ctx)`? (is there a context we can pipe down?)

Reply:
There's no context within [BubbleTea](https://github.com/search?q=repo%3Acharmbracelet%2Fbubbletea%20context&type=code), that said we can probably create a Context outside the TUI, and send it through the pipe down. 

It won't necessarily act the way a normal context would, so might be prudent to just use context.Background() for a Wait() :thinking: